src/main/java/org/jabref/gui/desktop/os/Linux.java

```diff
@@ -16,27 +18,48 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+@AllowedToUseAwt("Requires AWT to open a file with the native method")
 public class Linux implements NativeDesktop {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(Linux.class);
 
+    private void nativeOpenFile(String filePath) {
+        new Thread(() -> {
+            try {
+                File file = new File(filePath);
+                Desktop.getDesktop().open(file);
+                System.out.println("Open file in default application with Desktop integration");
+            } catch (IllegalArgumentException e) {
+                System.out.println("Fail back to xdg-open");
+                try {
+                    String cmd = "xdg-open " + filePath;
+                    Runtime.getRuntime().exec(cmd);
+                } catch (Exception e2) {
+                    System.out.println("Open operation not successful: " + e2);
+                }
+            } catch (IOException e) {
+                System.out.println("Native open operation not successful: " + e);
+            }
+        }).start();
+    }
+
     @Override
     public void openFile(String filePath, String fileType) throws IOException {
         Optional<ExternalFileType> type = ExternalFileTypes.getInstance().getExternalFileTypeByExt(fileType);
         String viewer;
 
         if (type.isPresent() && !type.get().getOpenWithApplication().isEmpty()) {
             viewer = type.get().getOpenWithApplication();
+            ProcessBuilder processBuilder = new ProcessBuilder(viewer, filePath);
```

Comment:
@Siedlerchr 
Can we not remove this part and replace it with
`Runtime.getRuntime().exec`?
The code would be a lot cleaner..both here and in the openFileWithApplication methods..

Reply:
This method is used when you have defined a specific app in JabRef (Manage external file types in the Settings dialog -> External programs)

At least we need this StreamGobblers which read the input and error streams, otherwise it could lead to hanging
https://www.infoworld.com/article/2071275/when-runtime-exec---won-t.html